### DISCUSSION TITLE
Source generator and Blazor components

### DISCUSSION THREAD

**Discussion by user72356** [dotnet/roslyn#55844] [Q&A] [answered]
In a source generator project, I'm unable to see the syntax trees of Blazor components source files (*.razor.g.cs). It seems like only my static .cs files are available in context.Compilation.SyntaxTrees, or in SyntaxReceivers.

My setup is quite simple and standard, I have a generator project:

<Project Sdk="Microsoft.NET.Sdk">

	<PropertyGroup>
		<TargetFramework>netstandard2.0</TargetFramework>
		<LangVersion>latest</LangVersion>
		<IsRoslynComponent>true</IsRoslynComponent>
	</PropertyGroup>

	<ItemGroup>
		<PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.11.0" PrivateAssets="all" />
		<PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.2" PrivateAssets="all" />
	</ItemGroup>

</Project>

And a Blazor Webassembly project:

<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">

	<PropertyGroup>
		<TargetFramework>net5.0</TargetFramework>
	</PropertyGroup>

	<ItemGroup>
		<PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="5.0.9" />
		<PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="5.0.9" PrivateAssets="all" />
		<PackageReference Include="System.Net.Http.Json" Version="5.0.0" />
	</ItemGroup>

	<ItemGroup>
		<ProjectReference Include="..\Generator\Generator.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
	</ItemGroup>

</Project>

I setup the generator project as the startup project and launching it as a Roslyn Component with F5. I hit my breakpoints properly, but none of the *.razor.g.cs are available to parse.

HELP!

**Comment by tmat** [comment]
@NTaylorMullen

**Comment by tmat** [comment]
Razor source generators currently only run during build and in very specific scenarios in the IDE such as Hot Reload.

**Comment by tmat** [comment, marked as answer]
Your source generator will be run in the IDE, but the Razor source generator that produces *.razor.g.cs won't.

**Reply by user72356** [reply]
I see. Any idea how I could properly debug my source generator within the IDE? I know of Debugger.Launch().

**Reply by Unskilledcrab** [reply]
> I see. Any idea how I could properly debug my source generator within the IDE? I know of Debugger.Launch().

![Screenshot_20210918-031327_Chrome.jpg](https://user-images.githubusercontent.com/12536115/133879983-6d614a57-c9e1-4a3d-a2b8-5f139d52b8c7.jpg)

I would highly recommend this method. You can make this a simple powershell script that you can run and then it will only build your project in that configuration when you run that script. This way the debugger isn't constantly popping up with the IDE runs its compilation process

**Reply by jasonmalinowski** [reply]
@Unskilledcrab: we also have fancier debugging support now: http://stevetalkscode.co.uk/debug-source-generators-with-vs2019-1610
